pocs/tests/test_filterwheel.py

```diff
@@ -0,0 +1,148 @@
+import pytest
+import math
+from timeit import timeit
+import time
+
+from astropy import units as u
+
+from pocs.filterwheel.simulator import FilterWheel as SimFilterWheel
+from pocs.camera.simulator import Camera as SimCamera
+
+
+@pytest.fixture(scope='module')
+def filterwheel():
+    sim_filterwheel = SimFilterWheel(filter_names=['one', 'deux', 'drei', 'quattro'],
+                                     move_time=0.1 * u.second,
+                                     timeout=0.5 * u.second)
+    return sim_filterwheel
+
+# intialisation
+
+
+def test_init(filterwheel):
+    assert isinstance(filterwheel, SimFilterWheel)
+    assert filterwheel.is_connected
+
+
+def test_camera_init():
+    sim_camera = SimCamera(filterwheel={'model': 'simulator',
+                                        'filter_names': ['one', 'deux', 'drei', 'quattro']})
+    assert isinstance(sim_camera.filterwheel, SimFilterWheel)
+    assert sim_camera.filterwheel.is_connected
+    assert sim_camera.filterwheel.uid
+    assert sim_camera.filterwheel.camera is sim_camera
+
+
+def test_camera_association_on_init():
+    sim_camera = SimCamera()
+    sim_filterwheel = SimFilterWheel(filter_names=['one', 'deux', 'drei', 'quattro'],
+                                     camera=sim_camera)
+    assert sim_filterwheel.camera is sim_camera
+
+
+def test_with_no_name():
+    with pytest.raises(ValueError):
+        sim_filterwheel = SimFilterWheel()
+
+# Basic property getting and (not) setting
+
+
+def test_model(filterwheel):
+    model = filterwheel.model
+    assert model == 'simulator'
+    with pytest.raises(AttributeError):
+        filterwheel.model = "Airfix"
+
+
+def test_name(filterwheel):
+    name = filterwheel.name
+    assert name == 'Simulated Filter Wheel'
+    with pytest.raises(AttributeError):
+        filterwheel.name = "Phillip"
+
+
+def test_uid(filterwheel):
+    uid = filterwheel.uid
+    assert uid.startswith('SW')
+    assert len(uid) == 6
+    with pytest.raises(AttributeError):
+        filterwheel.uid = "Can't touch this"
+
+
+def test_filter_names(filterwheel):
+    names = filterwheel.filter_names
+    assert isinstance(names, list)
+    for name in names:
+        assert isinstance(name, str)
+    with pytest.raises(AttributeError):
+        filterwheel.filter_names = ["Unsharp mask", "Gaussian blur"]
+
+# Movement
+
+
+def test_move_number(filterwheel):
+    assert filterwheel.position == 1
+    e = filterwheel.move_to(2)
+    assert math.isnan(filterwheel.position)  # position is NaN while between filters
```

Comment:
Should this move a larger position so there is no chance of some race-condition (not that there probably is anyway)